app/helpers/ExerciseConfigChecker.php

```diff
@@ -0,0 +1,69 @@
+<?php
+namespace App\Helpers;
+
+use App\Exceptions\ExerciseConfigException;
+use App\Helpers\ExerciseConfig;
+use App\Helpers\ExerciseConfig\Compilation\CompilationParams;
+use App\Model\Entity\Exercise;
+use App\Model\Entity\RuntimeEnvironment;
+use Nette\SmartObject;
+
+class ExerciseConfigChecker {
+  use SmartObject;
+
+  private $compiler;
+
+  private $validator;
+
+  private $loader;
+
+  public function  __construct(ExerciseConfig\Compiler $compiler, ExerciseConfig\Validator $validator, ExerciseConfig\Loader $loader) {
+    $this->compiler = $compiler;
+    $this->validator = $validator;
+    $this->loader = $loader;
+  }
+
+  /**
+   * Make up names of submitted files for a runtime environment.
+   * This is necessary because we do not have any real submissions yet, but the compiler needs their names.
+   * TODO when we implement a mechanism that ensures further constraints on submitted files, it must be reflected here
+   */
+  private function conjureSubmittedFiles(RuntimeEnvironment $environment) {
+    $extension = current($environment->getExtensionsList());
+    return ["main.{$extension}"];
+  }
+
+  public function check(Exercise $exercise) {
```

Comment:
Please comment this or whole class with appropriate description of what is done and how is exercise changed.